p4rt_app/utils/table_utility.cc

```diff
@@ -15,43 +15,86 @@
 
 #include "absl/status/status.h"
 #include "gutil/status.h"
+#include "absl/strings/substitute.h"
 #include "p4_pdpi/utils/annotation_parser.h"
 #include "swss/schema.h"
 
 namespace p4rt_app {
 namespace table {
 
+// This is what orch agent understands as fixed tables
+std::vector<std::string> FixedTables {
+  // Table names get added here runtime (based on schema for fixed tables)
+  // Additionally initialize this table now so existing build time tests pass
+  APP_P4RT_ROUTER_INTERFACE_TABLE_NAME
+};
+
 std::string TypeName(Type type) {
   switch (type) {
     case Type::kAcl:
       return "ACL";
     case Type::kFixed:
       return "FIXED";
-    case Type::kDefinition:
+    case Type::kExt:
+      return "EXT";
+    case Type::kAclDefinition:
       return APP_P4RT_ACL_TABLE_DEFINITION_NAME;
+    case Type::kTblsDefinitionSet:
+      return APP_P4RT_TABLES_DEFINITION_TABLE_NAME;
   }
   // TODO: return status failure.
 }
 
 absl::StatusOr<Type> TypeParse(absl::string_view type_name) {
   if (type_name == "ACL") return Type::kAcl;
   if (type_name == "FIXED") return Type::kFixed;
-  if (type_name == APP_P4RT_ACL_TABLE_DEFINITION_NAME) return Type::kDefinition;
+  if (type_name == "EXT") return Type::kExt;
+  if (type_name == APP_P4RT_ACL_TABLE_DEFINITION_NAME) {
+          return Type::kAclDefinition;
+  }
+  if (type_name == APP_P4RT_TABLES_DEFINITION_TABLE_NAME) {
+          return Type::kTblsDefinitionSet;
+  }
   return gutil::InvalidArgumentErrorBuilder()
          << "\"" << type_name << "\" does not name a valid table::Type.";
 }
 
 }  // namespace table
 
+void addSchemaSupportedTable(std::string table_name)
+{
+  absl::AsciiStrToUpper(&table_name);
+  table_name = "FIXED_" + table_name;
+
+  if (std::find(p4rt_app::table::FixedTables.begin(),
+                p4rt_app::table::FixedTables.end(),
+                table_name) == p4rt_app::table::FixedTables.end()) {
+    p4rt_app::table::FixedTables.push_back(table_name);
+  }
+}
+
 absl::StatusOr<table::Type> GetTableType(
     const pdpi::IrTableDefinition& ir_table) {
+  std::string table_name =
+              absl::Substitute("$0_$1", "FIXED", ir_table.preamble().alias());
+  absl::AsciiStrToUpper(&table_name);
   auto result =
       pdpi::GetAnnotationBody("sai_acl", ir_table.preamble().annotations());
+
   switch (result.status().code()) {
     case absl::StatusCode::kOk:
       return table::Type::kAcl;
     case absl::StatusCode::kNotFound:
-      return table::Type::kFixed;
+      if (std::find(p4rt_app::table::FixedTables.begin(),
+                    p4rt_app::table::FixedTables.end(),
+                    table_name) != p4rt_app::table::FixedTables.end()) {
+        return table::Type::kFixed;
+      }
+
+      // Return kExt table-type conditionally, until extension tables
+      //     validation is in place. Once that is in place,
+      //     return kExt table-type unconditionally
+      return CONDITIONAL_kExt(ir_table);
```

Comment:
As a general guideline in this dir, macros are not preferred. Can we inline that check here,
if (id > 0x02000000 & id < 0x03000000)
   return kFixed
else if (id > 0x03000000)
   return kExt

Reply:
sure, will change it to inline